From: Mathieu Baudier <mbaudier@argeo.org>
Date: Tue, 11 Mar 2025 15:40:11 +0100
Subject: use-llama-cpp-library

# Conflicts:
#	examples/talk-llama/CMakeLists.txt

# Conflicts:
#	examples/talk-llama/CMakeLists.txt
---
 examples/talk-llama/CMakeLists.txt | 37 ++++---------------------------------
 1 file changed, 4 insertions(+), 33 deletions(-)

diff --git a/examples/talk-llama/CMakeLists.txt b/examples/talk-llama/CMakeLists.txt
index 549842a..f86a338 100644
--- a/examples/talk-llama/CMakeLists.txt
+++ b/examples/talk-llama/CMakeLists.txt
@@ -2,42 +2,13 @@ if (WHISPER_SDL2)
     set(CMAKE_CXX_STANDARD 17)
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-    file(GLOB SRC_MODELS models/*.cpp)
-
     set(TARGET whisper-talk-llama)
-    add_executable(${TARGET} talk-llama.cpp
-        llama.cpp
-        llama-adapter.cpp
-        llama-arch.cpp
-        llama-batch.cpp
-        llama-chat.cpp
-        llama-context.cpp
-        llama-cparams.cpp
-        llama-grammar.cpp
-        llama-graph.cpp
-        llama-hparams.cpp
-        llama-impl.cpp
-        llama-io.cpp
-        llama-kv-cache.cpp
-        llama-kv-cache-iswa.cpp
-        llama-memory-recurrent.cpp
-        llama-memory-hybrid.cpp
-        llama-memory-hybrid-iswa.cpp
-        llama-memory.cpp
-        llama-mmap.cpp
-        llama-model-loader.cpp
-        llama-model-saver.cpp
-        llama-model.cpp
-        llama-quant.cpp
-        llama-sampler.cpp
-        llama-vocab.cpp
-        unicode.cpp
-        unicode-data.cpp
-        ${SRC_MODELS})
+    add_executable(${TARGET} talk-llama.cpp)
     target_include_directories(${TARGET} PRIVATE ${SDL2_INCLUDE_DIRS})
 
-    target_link_libraries(${TARGET} PRIVATE common common-sdl whisper ${SDL2_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
-    install(TARGETS ${TARGET} RUNTIME)
+    target_link_libraries(${TARGET} PRIVATE common common-sdl whisper llama ${SDL2_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+
+	install(TARGETS ${TARGET} RUNTIME)
 
     if(WIN32)
         # It requires Windows 8.1 or later for PrefetchVirtualMemory
